File: bestia-core/src/main/java/net/bestia/core/message/LogoutMessage.java
package net.bestia.core.message;

public class LogoutMessage extends Message {
	
	public static final String MESSAGE_ID = "system.logout";

	public LogoutMessage() {
		// no op.
	}
	
	public LogoutMessage(Message message) {
		setAccountId(message.getAccountId());
	}

	@Override
	public String getMessageId() {
		return MESSAGE_ID;
	}
}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/actor/socket/LogoutMessage.kt
package net.bestia.zoneserver.actor.socket

/**
 * This message is send to the player in order to signal a (forced) logout from
 * the system.
 *
 * @author Thomas Felix
 */
data class LogoutMessage(
    val state: LoginResponse = LoginResponse.DENIED_NO_REASON
)